File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        #bruteforce approach
        swaps = k
        for i in range(len(blocks)-k+1):
            swaps = min(swaps,blocks[i:i+k].count('W'))
        return swaps

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) 
    {
        //let's use two pointers
        int swaps = 0;
        for(int i = 0 ; i < k ; i++){if(blocks.charAt(i) == 'W')swaps++;}
        int currentSwaps = swaps;
        for(int i = 0, j = k; j < blocks.length();i++,j++)
        {
            if(blocks.charAt(i)=='W' && blocks.charAt(j)=='B')currentSwaps--;
            else if(blocks.charAt(i)=='B' && blocks.charAt(j) == 'W')currentSwaps++;
            swaps = Math.min(swaps,currentSwaps);
        }
        return swaps;
        
    }
}